lib/ClangImporter/ClangImporter.cpp

```diff
@@ -8774,3 +8781,18 @@ bool importer::isClangNamespace(const DeclContext *dc) {
 
   return false;
 }
+
+bool importer::isSymbolicCircularBase(const clang::CXXRecordDecl *symbolicClass,
```

Comment:
Should we only do this when we are importing something in symbolic mode?

Reply:
Yeah that's a good point. I'm not entirely sure whether we should, because the symbolic mode is only the trigger for this issue rather than the actual problem itself (which is that we're not handling `CxxRecordDecl`s that describe uninstantiated templates).

In any case, I can add the check just for this to be an extra conservative patch, since I presume we'll need to cherry-pick this.

Reply:
I think we should be consistent here in terms of symbolic vs non-symbolic mode, i.e. bailing in both modes, like this patch does currently.